snafu-derive/src/lib.rs

```diff
@@ -669,9 +669,19 @@ impl<'a> DisplayImpl<'a> {
                     ..
                 } = *variant;
 
-                let format = display_format
-                    .as_ref()
-                    .map_or_else(|| quote! { stringify!(#variant_name) }, |v| quote! { #v });
+                let format = display_format.as_ref().map_or_else(
+                    || match source_field {
+                        Some(source_field) => {
+                            let field_name = &source_field.name;
+                            // Format before using quote! to avoid adding extra whitespace between the
+                            // variant name and the colon.
+                            let format_string = format!("{}: {{}}", variant_name);
+                            quote! { #format_string, #field_name }
+                        }
+                        None => quote! { stringify!(#variant_name) },
+                    },
+                    |v| quote! { #v },
+                );
```

Comment:
This section might flow better as (pseudocode)

```rust
match (display_format, source_field) {
    (Some(v), _) => ...,
    (None, Some(f)) => ...,
    (None, None) => ...,
}
```

Since we support Rust 1.18, you will probably need to use the explicit dereferencing and `ref` keyword as appropriate.